touchforms/backend/touchcare.py

```diff
@@ -31,263 +33,56 @@
+        self.host = sessionvars['host']
+        self.domain = sessionvars['domain']
+        self.query_func = query_factory(self.host, self.domain, self.auth, 'raw')
+        self.query_url = get_restore_url({'as': self.username + '@' + self.domain, 'version': '2.0'})
 
-    def get_object_id(self, object):
-        raise NotImplementedError("subclasses must handle this")
+        if force_sync or self.needs_sync():
+            self.perform_ota_restore(restore)
 
-    def fetch_object(self, object_id):
-        raise NotImplementedError("subclasses must handle this")
+        self.last_sync = self.sandbox.getLastSync()
 
-    def load_all_objects(self):
-        raise NotImplementedError("subclasses must handle this")
+    def perform_ota_restore(self, restore=None):
 
-    def load_object_ids(self):
-        raise NotImplementedError("subclasses must handle this")
+        self.sandbox = UserDataUtils.getClearedStaticStorage(self.username)
 
-    @property
-    def objects(self):
-        if self.fully_loaded:
-            return self._objects
-        else:
-            self.load_all_objects()
-        return self._objects
-
-    def put_object(self, object):
-        object_id = self.get_object_id(object)
-        self._objects[object_id] = object
-
-    def read(self, record_id):
-        logger.debug('read record %s' % record_id)
-        try:
-            # record_id is an int, object_id is a guid
-            object_id = self.ids[record_id]
-        except KeyError:
-            return None
-        return self.read_object(object_id)
-
-    def read_object(self, object_id):
-        logger.debug('read object %s' % object_id)
-        if object_id not in self._objects:
-            self.put_object(self.fetch_object(object_id))
-        try:
-            return self._objects[object_id]
-        except KeyError:
-            raise Exception('could not find an object for id [%s]' % object_id)
-
-    def setReadOnly(self):
-        # todo: not sure why this exists. is it part of the public javarosa API?
-        pass
-
-    def getNumRecords(self):
-        return len(self.ids)
-
-    def iterate(self):
-        return StaticIterator(self.ids.keys())
-
-
-class CaseDatabase(TouchformsStorageUtility):
-
-    def get_object_id(self, case):
-        return case.getCaseId()
-
-    def fetch_object(self, case_id):
-        if ('case-id', case_id) in self.cached_lookups:
-            return self.cached_lookups[('case-id', case_id)][0]
-        return query_case(self.query_func, case_id)
-
-    def load_all_objects(self):
-        if self.form_context.get('cases', None):
-            cases = map(lambda c: case_from_json(c), self.form_context.get('cases'))
+        if not restore:
+            restore_xml = self.get_restore_xml()
+            text_file = open("restore.xml", "w")
+            text_file.write(restore_xml)
+            text_file.close()
+            restore_file = "restore.xml"
```

Comment:
Is it necessary to write this to file if we're just going to read it back again?